### PR TITLE
Operator syslog handle symbols

### PR DESCRIPTION
**Pull request by jsirianni**
## Description of Changes

Sometimes a syslog message with have a DOMAIN\username field, it reaches the syslog parser like this:
```
... user="observiq\\tom"] some message here
... user="observiq\\ron"] some message here
... user="observiq\\null"] some message here
```

[go-syslog](https://github.com/observIQ/go-syslog) will fail to parse log entries if these symbol characters are not escaped. 

- added handleSymbols() function that handles escaping these situations
- added test cases for handleSymbols, TestHandleSymbols
- added test cases to TestSyslogParser

Real symbols such as `\n`, `\t`, `\r` are not modified.

## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.7414031|+0.31032884|121.63605|-1.3840256|
|1|5000|6.603408|+1.5861058|135.0695|+4.9339905|
|1|10000|12.121335|+1.9831715|140.82085|-0.23168945|
|1|50000|55.897945|+4.9323044|173.96767|+0.3139801|
|1|100000|121.81668|+23.942162|230.9173|-2.0286865|
|10|100|2.2587185|+0.3448423|126.09402|-3.1774063|
|10|500|7.017511|+0.586421|133.16797|-1.3508911|
|10|1000|14.276174|+2.344985|141.81398|+0.85694885|
|10|5000|62.80613|+8.87447|182.01266|+2.2704773|
|10|10000|122.703514|+15.013123|216.89197|-6.207443|

File: operator/builtin/parser/syslog/syslog.go [hunk 1]
```diff
@@ -205,6 +208,14 @@ func toBytes(value interface{}) ([]byte, error) {
 	}
 }
 
+// handleSymbols escapes characters that appear to be symbol characters, but are not.
+// If not escaped properly, go-syslog will fail to parse the entry. Quotes that are
+// escaped will not be affected.
+func handleSymbols(b []byte) []byte {
+	b = bytes.Replace(b, []byte(`\`), []byte(`\\`), -1)
+	return bytes.Replace(b, []byte(`\\"`), []byte(`\"`), -1)
```
Comment (djaglowski) [inline operator/builtin/parser/syslog/syslog.go:216]: Am I correct in understanding that the net effect here is that all backslashes are escaped, except those that are being used to escape `"`?

File: operator/builtin/parser/syslog/syslog_test.go [hunk 2]
```diff
@@ -156,6 +156,89 @@ func TestSyslogParser(t *testing.T) {
 			entry.Info,
 			"info",
 		},
+		{
```
Comment (djaglowski) [inline operator/builtin/parser/syslog/syslog_test.go:159]: Can you add one or more tests that demonstrate expected behavior in cases where the backslash is used for purposes other than those identified as problems?

File: operator/builtin/parser/syslog/syslog.go (hunk 1 above)
Comment (jsirianni) [reply operator/builtin/parser/syslog/syslog.go:216]: Correct. We want to leave stuff like this alone:
```
kay="some \"escaped\" value"
```

File: operator/builtin/parser/syslog/syslog_test.go (hunk 2 above)
Comment (jsirianni) [reply operator/builtin/parser/syslog/syslog_test.go:159]: I added `RFC5424-escaped-quote` to TestSyslogParser.

input
```
... user="invalid user: \"null\"" ...
```
output:
```
"user": "invalid user: \"null\"",
```

TestHandleSymbols also showcases this behavior.

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.7586406|+0.3275664|124.58877|+1.5686951|
|1|5000|6.4484653|+1.4311633|134.47333|+4.3378143|
|1|10000|12.862694|+2.7245302|138.35817|-2.6943665|
|1|50000|56.48976|+5.5241203|173.83499|+0.18130493|
|1|100000|112.82706|+14.952538|232.13658|-0.80940247|
|10|100|2.51736|+0.6034838|126.56776|-2.7036667|
|10|500|7.0174904|+0.5864005|132.75848|-1.760376|
|10|1000|13.362217|+1.4310284|138.95663|-2.0003967|
|10|5000|61.41303|+7.481369|180.7733|+1.0311127|
|10|10000|124.2013|+16.51091|226.37299|+3.2735748|

**Review by djaglowski — APPROVED** [review]

**Pull request merged**